app/src/main/java/com/github/multimatum_team/multimatum/model/datetime_parser/DateTimePatternsGenerator.kt

```diff
@@ -103,53 +161,81 @@ class DateTimePatternsGenerator(private val currentDateProvider: () -> LocalDate
             timeFor(hour = args.getInt(0), 0)
         }
 
+    @PatternPair
     private val `15h00` =
         listOf(
             Token::asHour24,
-            Token::asTimeSeparator,
+            timeSeparator,
             Token::asMinute
         ) to { args: List<Any?> ->
             timeFor(hour = args.getInt(0), minute = args.getInt(2))
         }
 
+    @PatternPair
     private val `3am` =
         listOf(
             Token::asHour12,
-            tokenMatchingOneOf("am")
+            am
         ) to { args: List<Any?> ->
             timeFor(hour = args.getInt(0), minute = 0)
         }
 
+    @PatternPair
     private val `3pm` =
         listOf(
             Token::asHour12,
-            tokenMatchingOneOf("pm")
+            pm
         ) to { args: List<Any?> ->
             timeFor(hour = args.getInt(0) + 12, minute = 0)
         }
 
+    @PatternPair
+    private val `3h00am` =
+        listOf(
+            Token::asHour12,
+            timeSeparator,
+            Token::asMinute,
+            am
+        ) to { args: List<Any?> ->
+            timeFor(hour = args.getInt(0), minute = args.getInt(2))
+        }
+
+    @PatternPair
+    private val `3h00pm` =
+        listOf(
+            Token::asHour12,
+            timeSeparator,
+            Token::asMinute,
+            pm
+        ) to { args: List<Any?> ->
+            timeFor(hour = args.getInt(0) + 12, minute = args.getInt(2))
+        }
+
```

Comment:
not sur to understand why you have private attrivute for speciffically 3h pm and 3h am

Reply:
These are different patterns, in the "am" case one can take the given hour directly, in the "pm" case it is needed to add 12 (e.g. 5pm = 17h).